python_modules/dagster/dagster/_core/execution/plan/compute_generator.py

```diff
@@ -179,13 +179,25 @@ def _filter_expected_output_defs(
         (result,) if not isinstance(result, tuple) or is_named_tuple_instance(result) else result
     )
     asset_results = [x for x in result_tuple if isinstance(x, AssetResult)]
-    remove_outputs = [
-        r.get_spec_python_identifier(
-            asset_key=x.asset_key or context.op_execution_context.asset_key
-        )
-        for x in asset_results
-        for r in x.check_results or []
-    ]
+
+    if not asset_results:
+        return output_defs
+
+    check_names_by_asset_key = {}
+    for spec in context.op_execution_context.assets_def.check_specs:
```

Comment:
hmm are there cases where we don't have an AssetsDefinition?

Reply:
probably best to degrade gracefully than explode if it is `None`

Reply:
it'll raise 
```
raise DagsterInvalidPropertyError(
                f"Op '{self.op.name}' does not have an assets definition."
            )
```
so we won't just get a None error. I think we need to just throw that, otherwise we'll hit an output validation error in the next fn